scipy/sparse/csgraph/tests/test_shortest_path.py

```diff
@@ -311,3 +311,20 @@ def test_buffer(method):
     G = scipy.sparse.csr_matrix([[1.]])
     G.data.flags['WRITEABLE'] = False
     shortest_path(G, method=method)
+
+
+def test_NaN_warnings():
+    with pytest.warns(None) as record:
+        shortest_path(np.array([[0, 1], [np.nan, 0]]))
+    for r in record:
+        assert r.category is not RuntimeWarning
+
+def test_lil_matrix():
+    # Test that using lil sparse matrix do not cause error 
+    G_dense = np.array([[0, 3, 0, 0 ,0],
+                        [0, 0, -1, 0, 0],
+                        [0, 0, 0, 2, 0],
+                        [0, 0, 0, 0, 4],
+                        [0, 0, 0, 0, 0]])
+    G_lil = scipy.sparse.lil_matrix(G_dense)
```

Comment:
For completeness, could you loop over CSR, CSC, COO, and LIL format here?

Reply:
Are you sure about COO? Restrictions in `csgraph_to_dense` alows only CSR, CSC and LIL. However, the test would still pass with COO because Johnson is chosen for this graph.

And is there some cleaner way to iterate through this formats than copy paste?

Reply:
I've added only CSR and CSC tests.